api/list/service_test.go

```diff
@@ -0,0 +1,92 @@
+package list
+
+import (
+	"context"
+	"log"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/time"
+
+	"github.com/gojekfarm/albatross/pkg/helmcli"
+	"github.com/gojekfarm/albatross/pkg/helmcli/flags"
+)
+
+// To satisfy the client interface, we have to define all methods(NewUpgrade, NewInstaller) on the mock struct
+// TODO: Find a way to isolate interface only for upgrade
+type mockHelmClient struct{ mock.Mock }
+
+func (m *mockHelmClient) NewUpgrader(fl flags.UpgradeFlags) helmcli.Upgrader {
+	return m.Called().Get(0).(helmcli.Upgrader)
+}
+
+func (m *mockHelmClient) NewInstaller(fl flags.InstallFlags) helmcli.Installer {
+	return m.Called().Get(0).(helmcli.Installer)
+}
+
+func (m *mockHelmClient) NewLister(fl flags.ListFlags) helmcli.Lister {
+	return m.Called().Get(0).(helmcli.Lister)
+}
+
+type mockLister struct{ mock.Mock }
+
+func (m *mockLister) List(ctx context.Context) ([]*release.Release, error) {
+	args := m.Called(ctx)
+	if len(args) < 1 {
+		log.Fatalf("error while mocking response for list")
+	}
+	return args.Get(0).([]*release.Release), args.Error(1)
+}
+
+func TestShouldReturnValidResponseOnSuccess(t *testing.T) {
+	helmcli := new(mockHelmClient)
+	lic := new(mockLister)
+	service := NewService(helmcli)
+	ctx := context.Background()
+	req := Request{Flags: Flags{Deployed: true}}
+	helmcli.On("NewLister").Return(lic)
+	chartloader, err := loader.Loader("../testdata/albatross")
+	if err != nil {
+		panic("Could not load chart")
+	}
+
+	chart, err := chartloader.Load()
+	if err != nil {
+		panic("Unable to load chart")
+	}
+
+	releases := []*release.Release{
+		{
+			Name:      "test-release",
+			Namespace: "test-namespace",
+			Version:   1,
+			Info: &release.Info{
+				FirstDeployed: time.Now(),
+				Status:        release.StatusDeployed,
+			},
+			Chart: chart,
+		},
+	}
+
+	lic.On("List", ctx).Return(releases, nil)
+
+	resp, err := service.List(ctx, req)
+
+	assert.NoError(t, err)
+	require.NotNil(t, resp)
+	rel := resp.Releases[0]
```

Comment:
as a practice verify with `require` not nil or length > 0 before verifying fields